src/syscheckd/run_realtime.c

```diff
@@ -91,10 +90,15 @@ int realtime_checksumfile(const char *file_name)
             }
             send_syscheck_msg(alert_msg);
 
+            free(buf);
+
             return (1);
         } else {
             mdebug2("Discarding '%s': checksum already reported.", file_name);
         }
+
+        free(buf);
```

Comment:
This `free()` would deallocate memory already present in the hash table. The string should only be freed when the hash table is updated (`OSHash_Update()`) like in the first case.